src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -463,11 +463,10 @@ internal void ReadToken(ReadOnlyMemory<char> encodedTokenMemory)
                 // empty payload for JWE's {encrypted tokens}.
                 Payload = new JsonClaimSet();
 
-                if (Dot3 == encodedTokenSpan.Length) // TODO: Should this be encodedJsonSpan.Length - 1? 
-                    throw LogHelper.LogExceptionMessage(new ArgumentException(LogMessages.IDX14121));
```

Comment:
I think previously this line would never be hit. So changing the exception type to `SecurityTokenMalformedException` shouldn't be a breaking change.

Reply:
Agreed.